isocroft-syntax.md

```diff
@@ -0,0 +1,913 @@
+## SOURCE FILE
+- Source file can be empty, otherwise it must contain a valid set of (any 66536) UTF-8 codepoints.
+
+- Support for UTF-8 identifiers is planned.
+
+
+## COMMENTS
+#### Single-line comments
+- Single-line comments start with `#` and `//`
+    ```php
+    # Hello, World!
+    ```
+    ```java
+    // Hello, World!
+    ```
+
+    The singline comment starts with `#` so the plain source can be executed natively using shebang `(#!)` or system command can be executed at the start of the source file.
+
+#### Multiline comments
+- Nested multiline comment is allowed.
+    ```rust
+    /*
+     * Multiline comments can span multiple lines.
+     * They begin with `/*`.
+     * And they end with `*/`.
+     */
+    ```
+## DATA TYPES
+- Data types in ratio
+
+* int > a sequence of symbols from the standard number set (e.g. 2489)
+* long > two or more `int` type(s) put together as a sequence (e.g. 1267383994747489948947333344747)
+* float >
+* char > a single symbol from the standard chracter set ('$')
+* str > two or more `char` type(s) put together ('$rw32')
```

Comment:
I disagree with this approach to implementing strings. Modern languages support UTF-8 strings. UTF-8 strings aren't just sequence of characters because the character (actually codepoint) bit-width varies. They have a multi-byte encoding which uses between 1 and 4 bytes per character. 

Your char type cannot be 8-bit because it cannot represent the entire Unicode range, it can only represnt ASCII or extended ASCII charset. 

If your char type is 16-bit, then you char can represent UTF-16 (which is what Java did), but that is really limited as well, because Unicode 2019 has codepoints that can't fit in two bytes. So most languages tend to opt for 32-bit char types, because it covers the entire Unicode.

The issue with making your string just a sequence of char types is that the encoding is either going to be very wasteful or very limited. UTF-8 solves wastage issue and gives up O(1) access. The internet (heck the world) is in support of UTF-8.

In essence, string should be a UTF-8 encoding, not an array of characters.

I'd reserve this type of implementation for ascii-only string types or go-like rune types.

Reply:
Yes, you are very very correct! Which is why i included the `str` type against all restraint on my part. I wanted to discuss with everyone all the edge cases (as you have mentioned with code-points - especially code-points from Arab or German characters) with the rest of the team and seek out the best implementation. 

I believe you have just made enough case not to go with a UTF-16 or UTF-8 fixed byte encoding > multi-byte encoding perhaps